### PR TITLE
Directive irls

### PR DESCRIPTION
**Pull request by sgkang**
Efforts to make use of IRLS (or sparse inversion) for EM and DC inversions. 

- Fix_Jmatrix option: fix sensitivity matrix when updating for IRLS steps (after smooth inversion)
- BetaSearch: put an option that preventing betasearch

### CODE REVIEW

**Comment by sgkang** [comment]
@fourndo good point. simpegEM1D can be an example. I am pretty sure that I have customized function to evaluate JtJ for that case.

**Comment by sgkang** [comment]
Hi @fourndo and @lheagy can you approve (or review) this to be merged?

File: SimPEG/Directives.py [hunk 1]
```diff
@@ -972,13 +977,13 @@ class UpdateSensitivityWeights(InversionDirective):
...
     Directive to take care of re-weighting
     the non-linear magnetic problems.
-
     """
 
     mapping = None
     JtJdiag = None
     everyIter = True
     threshold = 1e-12
+    switch = True
```
Comment (lheagy) [inline SimPEG/Directives.py]: what does this do? is there a more descriptive name we could use?

File: SimPEG/Directives.py [hunk 2]
```diff
@@ -1073,7 +1073,7 @@ def getJtJdiag(self):
 
             m = self.invProb.model
 
-            self.JtJdiag += [mkvc(np.sum((dmisfit.W*prob.getJ(m)).power(2.), axis=0))]
+            self.JtJdiag += [mkvc(np.sum((dmisfit.W*prob.getJ(m))**(2.), axis=0))]
```
[the remark this answers is not in the document]
Comment (sgkang) [reply SimPEG/Directives.py]: I agree with you! That needs to be handled in the problem class.

File: SimPEG/Directives.py [hunk 3]
```diff
@@ -1037,13 +1043,15 @@ def getWr(self):
         """
 
         wr = np.zeros_like(self.invProb.model)
+        if self.switch:
```
[the remark this answers is not in the document]
Comment (sgkang) [reply SimPEG/Directives.py]: `activated` sounds better to me.

**Pull request merged**